flambe/field/text.py

```diff
@@ -138,62 +199,100 @@ def vocab_size(self) -> int:
             # Tokenize and add to vocabulary
             for token in self.tokenizer(example):
                 if token not in self.vocab:
-                    if model is not None:
-                        if token in model:
-                            self.vocab[token] = index = index + 1
-                            embeddings_matrix.append(torch.tensor(model[token]))
-                        else:
-                            if self.unk_init_all:
-                                # Give every OOV it's own embedding
-                                self.vocab[token] = index = index + 1
-                                embeddings_matrix.append(torch.randn(model.vector_size))
-                            else:
-                                # Collapse all OOV's to the same token
-                                # id
-                                self.vocab[token] = self.vocab[self.unk]
-                            self.unk_numericals.add(self.vocab[token])
+                    self.vocab[token] = index = index + 1
+
+    def _build_embeddings(self, model: KeyedVectors) -> Tuple[odict, torch.Tensor]:
+        """
+        Create the embeddings matrix and the new vocabulary in
+        case this objects needs to use an embedding model.
+
+        A new vocabulary needs to be built because of the parameters
+        that could allow, for example, collapsing OOVs.
+
+        Parameters
+        ----------
+        model: KeyedVectors
+            The embeddings
+
+        Returns
+        -------
+        Tuple[OrderedDict, torch.Tensor]
+            A tuple with the new embeddings and the embedding matrix
+        """
+        embedding_matrix: List[torch.Tensor] = []
+        new_vocab: odict[str, int] = odict()
+
+        new_index = -1
+
+        for token, index in self.vocab.items():
+            if token in self.specials:
+                emb = torch.tensor(model[token]) if \
+                    token in model else torch.randn(model.vector_size)
+                embedding_matrix.append(emb)
+                new_vocab[token] = new_index = new_index + 1
+            else:
+                if token in model:
+                    embedding_matrix.append(torch.tensor(model[token]))
+                    new_vocab[token] = new_index = new_index + 1
+                else:
+                    self.unk_numericals.add(self.vocab[token])
+
+                    if self.unk_init_all:
+                        embedding_matrix.append(torch.randn(model.vector_size))
+                        new_vocab[token] = new_index = new_index + 1
                     else:
-                        self.vocab[token] = index = index + 1
+                        # Collapse all OOV's to the same <unk> token id
+                        new_vocab[token] = new_vocab[self.unk]
+
+        if self.setup_all_embeddings:
+            for token in model.vocab.keys():
+                if token not in new_vocab:
+                    new_vocab[token] = index
```

Comment:
nit: this index setter is written differently than the rest of the file

Reply:
I think this should be `new_index`

Reply:
good catch. Done.